src/common/io.py

```diff
@@ -1344,17 +1344,10 @@ def _get_trajectory(self, data_index, start_index = 0, stop_index = None):
             nbr_variables  = self._data_2_info["nbr_variables"]
 
             # Account for sub-sets of data
-            if start_index > 0:
-                new_file_position = file_position + start_index*sizeof_type*nbr_variables
-                new_nbr_points = nbr_points - start_index
-            else:
-                new_file_position = file_position
-                new_nbr_points = nbr_points
-
-            if stop_index is not None and stop_index > 0:
-                new_nbr_points = stop_index
-                if start_index > 0:
-                    new_nbr_points -= start_index
+            start_index = max(0, start_index)
+            stop_index = max(0, nbr_points if stop_index is None else min(nbr_points, stop_index))
```

Comment:
One part of me thinks this should be `stop_index = max(0, nbr_points-1 if stop_index is None else min(nbr_points-1, stop_index))`, but tests only pass with implementation as above. Not sure where I am thinking incorrectly.

Reply:
Lets do a test. Create a test which have 100 result points. Try this method with using stop_index = 99,100,101. What do you get? What should you get?

Reply:
And is this what you would've expect?

Reply:
Yes